packages/types/src/index.ts

```diff
@@ -372,37 +375,258 @@ export interface Itinerary {
   walkDistance: number;
   walkLimitExceeded: boolean;
   walkTime: number;
-}
+};
+
+export type ElevationProfile = {
+  maxElev: number;
+  minElev: number;
+  points: number[];
+  traversed: number;
+  gain: number;
+  loss: number;
+};
 
 /**
  * Used to model a location that is used in planning a trip.
  */
-export interface Location {
+export type Location = {
   lat: number;
   lon: number;
-  name: string;
+  name?: string;
   /**
    * This is only used location that a user has saved. Can be either:
-   * "home" or "work"
+   * One of: 'home', 'work', 'stop' or 'recent'
    */
-  type?: string;
-}
+  type?: "home" | "work" | "stop" | "recent" | string;
+  category?: string;
+
+  /**
+   * This represents the last time that this location was selected in a
+   * search
+   */
+  timestamp?: number;
+
+  /* Sometimes used for displaying subinfo */
+  main?: string;
+  secondary?: string;
+};
 
 /**
  * Describes a transit stop entity to be rendered on the map.
  */
-export interface StopLayerStop extends LayerEntity {
+export type StopLayerStop = LayerEntity & {
   name: string;
-}
+};
 
 /**
- * @deprecated
+ * This models data about a stop and it's associated routes that is obtained
+ * from a transit index API.
+ */
+export type Stop = {
+  /**
+   * The stop code if the stop has one
+   */
+  code: string;
+  dist: number;
```

Comment:
Is this optional too?

Reply:
I think so. Refactored in db1658fad93479daa48f7229371b9aef2200ae86